File: backend/models/weather.py
from pydantic import BaseModel


class WeatherData(BaseModel):
    condition: str = "clear"
    temp_celsius: float = 30.0
    humidity: int = 60
    wind_kph: float = 10.0
    rain_probability: int = 0
    rain_mm_next_hour: float = 0.0
    fetched_at: str | None = None

    @property
    def is_rainy(self) -> bool:
        return self.rain_probability > 50 or self.condition in ("rain", "drizzle", "thunderstorm")

File: src/lib/api/weather.ts
import type { WeatherData, ApiResponse } from '@/types';

export async function getWeather(stationId: string): Promise<ApiResponse<WeatherData>> {
  const res = await fetch(`/api/weather/${stationId}`);
  if (!res.ok) return { data: null, error: 'Failed to fetch weather' };
  const json = await res.json();
  return {
    data: {
      condition: json.condition,
      tempCelsius: json.temp_celsius,
      humidity: json.humidity,
      windKph: json.wind_kph,
      rainProbability: json.rain_probability,
      rainMmNextHour: json.rain_mm_next_hour,
      fetchedAt: json.fetched_at,
    },
    error: null,
  };
}
